File: dev/app/scripts/NativeBridge/ScrollBar.ts
/// <reference path="../../modules/include/frameworks.d.ts" />

module NativeBridgeDevBed {
    export module NativeBridge {

        import Promise = CDP.NativeBridge.Promise;

        var TAG: string = "[NativeBridgeDevBed.NativeBridge.ScrollBar] ";

        /**
         * @class ScrollBar
         * @brief Native Scroll Bar をコントロールするブリッジクラス
         */
        export class ScrollBar extends CDP.NativeBridge.Gate {
            /**
             * constructor
             *
             */
            constructor() {
                super({
                    name: "ScrollBar",
                    android: {
                        packageInfo: "com.sony.cdp.sample.ScrollBar",
                    },
                    // iOS 版は不要
                });
            }

            ///////////////////////////////////////////////////////////////////////
            // public methods

            /**
             * Vertical ScrollBar の 表示
             */
            public showVertical(): void {
                super.exec("showVertical", <any>arguments);
            }

            /**
             * Vertical ScrollBar の 表示
             */
            public hideVertical(): JQueryPromise<void> {
                var df = $.Deferred<void>();
                super.exec("hideVertical", <any>arguments)
                    .always(() => {
                        df.resolve();
                    });
                return df.promise();
            }
        }
    }
}

File: dev/platforms/android/src/com/sony/cdp/sample/ScrollBar.java
package com.sony.cdp.sample;

import java.lang.reflect.InvocationTargetException;

import android.util.Log;

import com.sony.cdp.plugin.nativebridge.Gate;


/**
 * \~english
 * @class ScrollBar
 * @brief Scroll Bar management class.
 *
 * \~japanese
 * @class ScrollBar
 * @brief Scroll Bar 管理クラス
 */
public class ScrollBar extends Gate {
    private static final String TAG = "[com.sony.cdp.sample][Native][ScrollBar] ";


    ///////////////////////////////////////////////////////////////////////
    // public mehtods

    /**
     * \~english
     * Show vertical scrollBar.
     *
     * \~japanese
     * Vertical ScrollBar の 表示
     */
    public void showVertical() {
        try {
            Utils.getViewFromCordovaWebView(webView).setVerticalScrollBarEnabled(true);
        } catch (NoSuchMethodException e) {
            Log.e(TAG, "NoSuchMethodException", e);
        } catch (IllegalAccessException e) {
            Log.e(TAG, "IllegalAccessException", e);
        } catch (IllegalArgumentException e) {
            Log.e(TAG, "IllegalArgumentException", e);
        } catch (InvocationTargetException e) {
            Log.e(TAG, "InvocationTargetException", e);
        }
        Log.d(TAG, "showVertical() called.");
    }

    /**
     * \~english
     * Hide vertical scrollBar.
     *
     * \~japanese
     * Vertical ScrollBar を非表示
     */
    public void hideVertical() {
        try {
            Utils.getViewFromCordovaWebView(webView).setVerticalScrollBarEnabled(false);
        } catch (NoSuchMethodException e) {
            Log.e(TAG, "NoSuchMethodException", e);
        } catch (IllegalAccessException e) {
            Log.e(TAG, "IllegalAccessException", e);
        } catch (IllegalArgumentException e) {
            Log.e(TAG, "IllegalArgumentException", e);
        } catch (InvocationTargetException e) {
            Log.e(TAG, "InvocationTargetException", e);
        }
        Log.d(TAG, "hideVertical() called.");
    }
}
